back/genhtml.py

```diff
@@ -31,16 +33,106 @@ def points(game):
     mod = len(game[3:])
     return (spiel + mod) * colour
 
+
 def compute_scores(games):
-    scores = {}
-    for game in games:
-        score = points(game["game"]) * (-2, 1)[int(game["win"])]
-        scores[game["player"]] = scores.get(game["player"], 0) + score
+    games = list(games)
+    all_players = {game["player"] for game in games}
+
+    old_score_factor = 0.5
+    new_score_factor = 1
+
+    # Get month of first game
+    first_timestamp = date.fromtimestamp(int(games[0]["timestamp"]))
+    first_year = first_timestamp.year
+    first_month = first_timestamp.month
+
+    # Get current month
+    current_timestamp = datetime.now()
+
+    current_year = current_timestamp.year
+    current_month = current_timestamp.month
+
+    # Get all months since the first game
+    # [(2020, 1), (2020, 2), …]
+    all_months_in_years = []
+    for year in range(first_year, current_year + 1):
+        start_month = first_month if year == first_year else 1
+        end_month = current_month if year == current_year else 12
+        for month in range(start_month, end_month + 1):
+            all_months_in_years.append((year, month))
+
+    # Group all games by their month
+    # {(2020,1): [game_1, game_2], …}
+    games_grouped_by_month = {k: list(v) for k, v in itertools.groupby(
+        games,
+        key=lambda x: (
+            date.fromtimestamp(int(x["timestamp"])).year,
+            date.fromtimestamp(int(x["timestamp"])).month
+        ))}
+
+    # {name: (score, total_score, months)}
+    scores = {name: (0, 0, 0) for name in all_players}
+
+    for year, month in all_months_in_years:
```

Comment:
This whole computation seems a bit inefficient, but regarding to the size of our user base, I don't mind.